koku/reporting/provider/gcp/models.py

```diff
@@ -949,3 +969,853 @@ class Meta:
+
+    service_alias = models.CharField(max_length=256, null=True, blank=True)
+
+    account_id = models.CharField(max_length=50, null=False)
+
+    invoice_month = models.CharField(max_length=256, null=True, blank=True)
+
+    credit_amount = models.DecimalField(max_digits=24, decimal_places=9, null=True, blank=True)
+
+
+class GCPComputeSummaryByAccountP(models.Model):
+    """A summarized partitioned table specifically for UI API queries.
+
+    This table gives a daily breakdown of total cost by service and instance type.
+
+    """
+
+    class PartitionInfo:
+        partition_type = "RANGE"
+        partition_cols = ["usage_start"]
+
+    class Meta:
+        """Meta for GCPComputeSummaryByAccountP."""
+
+        db_table = "reporting_gcp_compute_summary_by_account_p"
+        indexes = [
+            models.Index(fields=["account_id"], name="gcpcompsumm_acc_account_id"),
+            models.Index(fields=["usage_start"], name="gcpcompsumm_acc_usage_start"),
+            models.Index(fields=["instance_type"], name="gcpcompsumm_acc_insttyp"),
+            models.Index(fields=["invoice_month"], name="gcpcompsumm_acc_invmonth"),
+        ]
+
+    id = models.UUIDField(primary_key=True)
+
+    usage_start = models.DateField(null=False)
+
+    usage_end = models.DateField(null=False)
+
+    instance_type = models.CharField(max_length=50, null=True)
+
+    usage_amount = models.DecimalField(max_digits=24, decimal_places=9, null=True)
+
+    unit = models.CharField(max_length=63, null=True)
+
+    unblended_cost = models.DecimalField(max_digits=24, decimal_places=9, null=True)
+
+    markup_cost = models.DecimalField(max_digits=24, decimal_places=9, null=True)
+
+    currency = models.CharField(max_length=10)
+
+    source_uuid = models.ForeignKey(
+        "api.Provider", on_delete=models.CASCADE, unique=False, null=True, db_column="source_uuid"
+    )
+
+    account_id = models.CharField(max_length=50, null=False)
+
+    invoice_month = models.CharField(max_length=256, null=True, blank=True)
+
+    credit_amount = models.DecimalField(max_digits=24, decimal_places=9, null=True, blank=True)
+
+
+class GCPComputeSummaryByRegionP(models.Model):
+    """A summarized partitioned table specifically for UI API queries.
+
+    This table gives a daily breakdown of total cost by service and instance type.
+
+    """
+
+    class PartitionInfo:
+        partition_type = "RANGE"
+        partition_cols = ["usage_start"]
+
+    class Meta:
+        """Meta for GCPComputeSummaryByRegionP."""
+
+        db_table = "reporting_gcp_compute_summary_by_region_p"
+        indexes = [
+            models.Index(fields=["account_id"], name="gcpcompsumm_reg_account_id"),
+            models.Index(fields=["usage_start"], name="gcpcompsumm_reg_usage_start"),
+            models.Index(fields=["instance_type"], name="gcpcompsumm_reg_insttyp"),
+            models.Index(fields=["invoice_month"], name="gcpcompsumm_reg_invmonth"),
+        ]
```

Comment:
For the ComputeServiceByRegion should we have an index on region?

Reply:
So, I agree with you, but I think we are gonna end up getting rid of these tables later on: https://issues.redhat.com/browse/COST-2106